WT-6870 Rename history store statistic to better describe its use case.

luke-pearson opened the pull request:
…ct actual work done

quchenhao reviewed (approved):
lgtm

quchenhao commented on src/history/hs_rec.c:
```diff
@@ -1066,6 +1062,9 @@ __hs_delete_key_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_
 
+        WT_STAT_CONN_INCR(session, cache_hs_key_truncate_mix_ts);
+        WT_STAT_DATA_INCR(session, cache_hs_key_truncate_mix_ts);
+
```
This is out of the scope of the pr. Should we have another stats to track the number of reinsert in the history store?

luke-pearson replied:
Having one stat to count deletions and one stat to count reinsertion will help us differentiate the scenario where we are truncating due to a key being deleted and otherwise. 

I think its worth doing that in this ticket. I'll have to reword both statistics.

luke-pearson replied:
On second thought, there's a stat in this function: cache_hs_key_truncate which already tracks the real work done. The work here will be to instead rename that statistic that implies mixed timestamps are in use here, when they don't actually need to be and keep the increment in the same location.

luke-pearson commented:
@quchenhao I've renamed the statistic instead in this PR and deleted the test, I think thats acceptable for what i'm trying to achieve in that I don't want people misled thinking that there are mixed mode timestamps happening when this can happen without timestamps.

quchenhao reviewed (approved):
lgtm

kommiharibabu reviewed (approved):
LGTM
